_layouts/page.html

```diff
@@ -68,6 +68,13 @@
             </div>
           {% endunless %}
         {% endif %}
+        {% if page.title == 'Glossary' %}
+          <div class="btn-group" role="group">
+          {% for letter in page.categories %}
+            <button type="button" class="btn btn-default"><a href="#{{ letter }}">{{ letter | upcase }}</a></button>
+          {% endfor %}
+          </div>
+        {% endif %}
```

Comment:
Or, if you just want to reuse this glossary button stuff, create another layout called `glossary` that uses the `page` layout and adds the buttons.